docs/optimizations/FileLocalization.md

```diff
@@ -0,0 +1,52 @@
+# The 'localization_optional' Optimization
+
+## Scope
+
+The 'localization_optional' optimization applies to `File` and `File?` inputs on `task`s. It allows
+you to save time and money by perhaps not localizing some `File`s to the execution environment.
+
+## Effect
+
+The optimization signals to Cromwell that the task has been written in such a way that:
+ * The task will still work if the file were to be localized - say from a cloud bucket to the task's execution environment; but...
+ * The task will *also* work if the file is *not* localized prior to execution - and remains a path to a value in an object store, for example.  
+
+Therefore, Cromwell will (usually) choose not to localize this file, in order to save time and money on disk size and `File` localization.
```

Comment:
When will Cromwell choose to localize the file, even when "localization_optional" is enabled?